grails-app/controllers/au/org/ala/biocollect/BioActivityController.groovy

```diff
@@ -1872,6 +1878,10 @@ class BioActivityController {
             model.error = "No project associated with the activity"
         } else if (projectService.isUserAdminForProject(userId, projectId) || activityService.isUserOwnerForActivity(userId, activity?.activityId)) {
             model = [activity: activity]
+            if (includeSiteData) {
+                model = activityAndOutputModel(activity, activity.projectId)
```

Comment:
we don't want to add all the information supplied by activityAndOutputModel. Instead add a property site with values name, siteId and geoJSON. Use siteService.get(model.activity.siteId, [view: 'brief', version: version]).

Reply:
Hi @temi , do I need to add this as well?
model.mapFeatures = model.site ? siteService.getMapFeatures(model.site) : []

Reply:
No. The output should look like below.
```
{
"activity" : {
  "activityId": "...",
  "site": {
    "name": "...",
    "siteId": "...",
    "geoJson": {}
  }
}
}
```